src/typecode/contenttype.py

```diff
@@ -287,6 +289,11 @@ def is_archive(self):
         """
         # FIXME: we should use extracode archive detection
         # TODO: also treat file systems as archives
+        #adding extract code detector
+        can_extract(location)
```

Comment:
Since you are fixing things, remove the line
`# FIXME: we should use extracode archive detection`

Also the comment `#adding extract code detector` is not needed

And calling `can_extract(location)` without assigning to a variable does not make sense.